diff --git a/dist/edge/index.js b/dist/edge/index.js
index 55d5aaabb7f76c338498f01e31d13e54561113d0..c1fac12d90d253c6dc87d99e4af1f6a6fe43f086 100644
--- a/dist/edge/index.js
+++ b/dist/edge/index.js
@@ -1412,6 +1412,7 @@ var buildJwt = async (credentials) => {
 };
 async function generateAuthToken(credentials) {
   try {
+    if(credentials.cfApiKey) return credentials.cfApiKey;
     const creds = credentials || await loadCredentials();
     const jwt = await buildJwt(creds);
     const response = await fetch("https://oauth2.googleapis.com/token", {
